src/main/java/com/google/sps/servlets/QueryServlet.java

```diff
@@ -46,16 +48,27 @@
     private final Gson gson = new GsonBuilder().create();
     private Response response;
     private User user;
+    private UrlOpener urlOpener;
+
+    public QueryServlet(UrlOpener opener){
+        this.urlOpener = opener;
+    }
 
+    public QueryServlet(){
+        this.urlOpener = new UrlOpener();
+    }
+    
     @Override
     // TODO: return a user-friendly error rather than throwing an exception
     public void doGet(HttpServletRequest servletRequest, HttpServletResponse servletResponse) throws IOException {
         HttpSession session = servletRequest.getSession(false);
         Response response = (Response) session.getAttribute("response");
-        if(response == null)
-            servletResponse.getWriter().println(gson.toJson(new Response("NO_RESULTS", null)));
-        else if (response.getStatus().equals("OK"))
+        if(response == null){
+            response = new Response("NO_RESULTS", new ArrayList<Restaurant>());
```

Comment:
make sure changing this to an empty list doesn't impact anything calling this method. null != an empty arraylist

Reply:
A null response will produce an error in the front end when we try to index it. This is meant to change the status to NO_RESULTS because the JS has an condition to handle an invalid response with no results in it